server/server.go

```diff
@@ -83,26 +57,117 @@ func (c *PreFilterServer) GetContentPackages(ctx context.Context, request api.Ge
 	defer span.End()
 
 	// TODO: user will be needed in spicedb queries -- set Authorization request header to the userid
-	if user, found := getUserFromContext(ctx); found {
-		fmt.Printf("user found in request: %s\n", user)
+
+	user, accountId, found := getIdentityFromContext(ctx)
+	if found {
+		fmt.Printf("indentity found in request: %s %d\n", user, accountId)
 	}
 
 	_, spiceSpan := c.Tracer.Start(ctx, "SpiceDB pre-filter call")
-	time.Sleep(time.Second) // mimics the delay calling out to SpiceDB
+
+	lrClient, err := c.SpicedbClient.LookupResources(ctx, &v1.LookupResourcesRequest{
+		ResourceObjectType: "inventory/host",
+		Permission:         "read",
+		Subject: &v1.SubjectReference{
+			Object: &v1.ObjectReference{
+				ObjectType: "user",
+				ObjectId:   "test",
+			},
+		},
+	})
+
+	if err != nil {
+		fmt.Errorf("spicedb error: %v", err)
+		return nil, err
+	}
+
+	var hostIDs []string
+	for {
+		next, err := lrClient.Recv()
+		if e.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			fmt.Errorf("spicedb error: %v", err)
+			return nil, err
+		}
+
+		hostIDs = append(hostIDs, next.GetResourceObjectId()) // e.g. service or inventory group
+	}
+
 	spiceSpan.End()
 
 	_, pgSpan := c.Tracer.Start(ctx, "Postgres query")
-	time.Sleep(time.Second) // mimics the delay calling out to Postgres
+
+	packageAccountData := make([]cachecontent.PackageAccountData, 0)
+	countError := packagesByHostIDs(&packageAccountData, accountId, hostIDs)
+	if countError != nil {
+		return nil, countError
+	}
+
+	packages, err := c.GetPackagesPayload(packageAccountData)
+
 	pgSpan.End()
 
-	packages, err := c.GetPackagesPayload()
 	return packages, err
 }
 
-func getUserFromContext(ctx context.Context) (user string, found bool) {
-	if user, ok := ctx.Value("user").(string); ok && user != "" {
-		return user, true
+func getIdentityFromContext(ctx context.Context) (user string, rhAccount int64, found bool) {
+	// Retrieve the value from the context
+	val := ctx.Value("user")
+	if val == nil {
+		// Handle case where the value is not found
+		return "", 0, false
+	}
+
+	// Convert the value to a string and split it
+	userInfo, ok := val.(string)
```

Comment:
I thought that the type assertion `user, ok := ctx.Value("user").(string)` tested both the case where `ctx.Value("user")` was nil and also when it was not a string (so only one if statement needed).

Reply:
yes this is correct

Reply:
fixed